include/mbedtls/config.h

```diff
@@ -2105,6 +2106,19 @@
  */
 #define MBEDTLS_HMAC_DRBG_C
 
+/**
+ * \def MBEDTLS_NIST_KW_C
+ *
+ * Enable the Key Wrapping mode for 128-bit block cipher,
+ * As defined in NIST SP800-38F
```

Comment:
Why is this PR restricted to AES, anyway? Why not support Camellia and ARIA as well? All it would take is a few more cases in a switch, and some test cases, right?

This is a general-purpose cryptographic library. We don't restrict the algorithms to NIST-approved ones.

Reply:
We can add those later as needed. This PR doesn't have to support (and tests, etc.) for those other algorithms. Implementing them now in this PR, especially as they aren't being requested yet, would be a small distraction. I'd be happy to see a follow up PR to add that support later, though.

Reply:
If we want to add these algorithms, then we shouldn't call this module `NIST_KW`.
NIST defines *only* AES as supported.
In addition,  RFC 3394 and RFC 5649 define AES Key wrapping.

Reply:
No. NIST defines a block cipher mode for key wrapping. We can use it for any cipher for which the formula works. It's the same as allowing e.g. CBC with Camellia: NIST doesn't allow Camellia, but we do.

We're implementing a feature and we follow a standard for that feature which happens to be published by NIST. This doesn't prevent us from including features that are not codified by NIST standards. If we wanted to make a library that's compliant to FIPS 140, we would need to have a mode where only NIST-Approved algorithms are allowed, which would restrict KW and KWP to AES, but this is not what we're doing here (and even FIPS 140 allows the code for other cryptographic algorithms to be present, as long as they aren't used in compliant mode, but that's too long a story for this venue).

Reply:
We can't say wee are NIST compliant for this algorithm if NIST specifically allows AES only.
But, we are getting off topic in this discussion.

Reply:
We can say we are NIST-compliant if we support what NIST wants. We don't have to exclude support for algorithms that NIST doesn't approve. Not even for a FIPS 140 certification!